test/test_matmul_scheduler.cpp

```diff
@@ -406,24 +408,1801 @@ TEST_F(MatmulSchedulerTest, EpilogueAlphaOutputCast_CUDA) {
+  NVFUSER_TEST_CUDA_ARCH_RANGE_GUARD(8, 0, 9, 0);
+  const auto layout = MatmulLayout::TT;
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  // A - tv0, B - tv1, C - tv2
+  auto tv0 = makeContigTensor(2, DataType::Half);
+  auto tv1 = makeContigTensor(2, DataType::Half);
+  auto tv2 = makeContigTensor(1, DataType::Half);
+
+  // tv3 := A x B
+  auto tv3 = matmul(tv0, tv1, layout, true);
+  auto tv4 = castOp(DataType::Half, tv3);
+
+  // tv5 := (A x B) + bias
+  auto tv5 = biasEpilogue(tv4, tv2);
+
+  // tv6 := relu((A x B) + bias)
+  auto tv6 = castOp(DataType::Half, relu(tv5));
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addInput(tv2);
+  fusion->addOutput(tv5);
+  fusion->addOutput(tv6);
+
+  NVF_CHECK(
+      1 == ir_utils::getOpsOfType<MmaOp>(fusion.get()).size(),
+      "matmul fusion must have at least one MmaOp");
+  NVF_CHECK(
+      ir_utils::getOpsOfType<MmaOp>(fusion.get()).front()->layout().has_value(),
+      "input layout has not be set for MmaOp");
+  NVF_CHECK(
+      MatmulLayout::TN ==
+          ir_utils::getOpsOfType<MmaOp>(fusion.get()).front()->layout().value(),
+      "the MmaOp layout of Ampere MMA must always be TN");
+
+  const auto fusion_layout = mma_utils::getMatmulLayout(fusion.get());
+  NVF_CHECK(
+      fusion_layout.isValid(),
+      "failed to get decide matmul layout through fusion definition");
+  NVF_CHECK(
+      fusion_layout.getData() == layout,
+      "mismatch between test layout (",
+      toString(layout),
+      ") and layout inferred from fusion definition (",
+      toString(fusion_layout.getData()),
+      ")");
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+
+  const int M = 504, N = 136, K = 248;
+
+  at::manual_seed(0);
+  auto t0 = matmulAtInput(layout, TensorMatmulPos::A, at::kHalf, M, N, K);
+  auto t1 = matmulAtInput(layout, TensorMatmulPos::B, at::kHalf, M, N, K);
+  auto t2 = matmulAtInput(layout, TensorMatmulPos::Bias, at::kHalf, M, N, K);
+
+  auto t3 = atMatmul(t0.to(at::kFloat), t1.to(at::kFloat), layout);
+  auto t4 = t3.to(at::kHalf);
+  auto t5 = atBiasEpilogue(t4, t2);
+  auto t6 = at::relu(t5);
+
+  auto outputs = executor_cache.runFusionWithInputs({t0, t1, t2});
+
+  NVF_CHECK(
+      !executor_cache.getMostRecentKernelRuntime()->isSegmented(),
+      "segmentation did happen");
+
+  // NOTE: increasted absolute tolerance to silence false negative verification
+  //       caused by different way of calculating reference
+  // D tensor results
+  NVF_CHECK(outputs[0].allclose(t5, 0.001, 0.001));
+  // Aux tensor results
+  NVF_CHECK(outputs[1].allclose(t6, 0.001, 0.001));
+}
+
+// Matmul test that uses segmenter for fusion for Ampere:
+//  D = (A x B) + bias
+//  Aux = relu(D)
+//  Precisions:
+//   - A (bf16 - t)
+//   - B (bf16 - t)
+//   - math (fp32 - s)
+//   - D (bf16 - t)
+//   - bias (bf16 - t)
+//   - aux (bf16 - t)
+TEST_F(MatmulSchedulerTest, DISABLED_EpilogueBiasReluAuxTST_CUDA) {
```

Comment:
Maybe TSS?

Reply:
Added in the second commit.

Thanks for suggestion!